packages/interaction-tracking/src/InteractionTracking.js

```diff
@@ -201,89 +184,77 @@ export function wrap(
 
   const wrappedInteractions = interactionsRef.current;
 
-  if (enableInteractionTrackingObserver) {
-    const subscriber = subscriberRef.current;
-    if (subscriber !== null) {
-      subscriber.onWorkScheduled(wrappedInteractions, threadID);
-    }
-
-    // Update the pending async work count for the current interactions.
-    // Update after calling subscribers in case of error.
-    wrappedInteractions.forEach(interaction => {
-      interaction.__count++;
-    });
+  let subscriber = subscriberRef.current;
+  if (subscriber !== null) {
+    subscriber.onWorkScheduled(wrappedInteractions, threadID);
   }
 
+  // Update the pending async work count for the current interactions.
+  // Update after calling subscribers in case of error.
+  wrappedInteractions.forEach(interaction => {
+    interaction.__count++;
+  });
+
   const wrapped = () => {
     const prevInteractions = interactionsRef.current;
     interactionsRef.current = wrappedInteractions;
 
-    if (enableInteractionTrackingObserver) {
-      const subscriber = subscriberRef.current;
+    subscriber = subscriberRef.current;
 
-      try {
-        let returnValue;
+    try {
+      let returnValue;
 
-        try {
-          if (subscriber !== null) {
-            subscriber.onWorkStarted(wrappedInteractions, threadID);
-          }
-        } finally {
-          try {
-            returnValue = callback.apply(undefined, arguments);
-          } finally {
-            interactionsRef.current = prevInteractions;
-
-            if (subscriber !== null) {
-              subscriber.onWorkStopped(wrappedInteractions, threadID);
-            }
-          }
+      try {
+        if (subscriber !== null) {
+          subscriber.onWorkStarted(wrappedInteractions, threadID);
         }
-
-        return returnValue;
       } finally {
-        // Update pending async counts for all wrapped interactions.
-        // If this was the last scheduled async work for any of them,
-        // Mark them as completed.
-        wrappedInteractions.forEach(interaction => {
-          interaction.__count--;
+        try {
+          returnValue = callback.apply(undefined, arguments);
```

Comment:
Just noticed this... `wrapped` is an arrow function here, so these are the arguments to the outer function. Need a test.

Reply:
Good catch.